apps/auth-app/src/components/sign-up/index.tsx

```diff
@@ -1,8 +1,269 @@
+import { useTranslation } from 'react-i18next';
+
+import DS from '@akashaproject/design-system';
+import getSDK from '@akashaproject/awf-sdk';
+import { useInjectedProvider } from '@akashaproject/ui-awf-hooks';
+
 import { RootComponentProps } from '@akashaproject/ui-awf-typings';
 
+import { StepOne } from './steps/StepOne';
+import { StepTwo } from './steps/StepTwo';
+import { StepThree } from './steps/StepThree';
+import { StepFour } from './steps/StepFour';
+
+const { Box, SignUpCard } = DS;
+
+export interface IInviteTokenForm {
+  submitted: boolean;
+  submitting: boolean;
+  success: boolean;
+  hasError: boolean;
+  errorMsg: string;
+}
+
 const SignUp: React.FC<RootComponentProps> = _props => {
-  return <div>Sign Up</div>;
+  const [activeIndex, setActiveIndex] = React.useState<number>(3);
+  const [inviteToken, setInviteToken] = React.useState<string>('');
+  const [inviteTokenForm, setinviteTokenForm] = React.useState<IInviteTokenForm>({
+    submitted: false,
+    submitting: false,
+    success: false,
+    hasError: false,
+    errorMsg: '',
+  });
+
+  const getInjectedProviderQuery = useInjectedProvider();
+  const injectedProvider = getInjectedProviderQuery.data;
+
+  const { t } = useTranslation();
+  const sdk = getSDK();
+
+  const DEFAULT_TOKEN_LENGTH = 24;
+
+  const handleIconClick = () => {
+    if (activeIndex === 0) return;
+    setActiveIndex(prev => prev - 1);
+  };
+
+  const handleNextStep = () => {
+    setActiveIndex(prev => prev + 1);
+  };
+
+  const onInputTokenChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setinviteTokenForm({
+      submitted: false,
+      submitting: false,
+      success: false,
+      hasError: false,
+      errorMsg: '',
+    });
+    setInviteToken(e.target.value);
+  };
+
+  const checkIsValidToken = () => {
+    setinviteTokenForm({
+      submitted: false,
+      submitting: true,
+      success: false,
+      hasError: false,
+      errorMsg: '',
+    });
+    lastValueFrom(sdk.api.auth.validateInvite(inviteToken))
+      .then(() => {
+        setinviteTokenForm({
+          submitted: true,
+          submitting: false,
+          success: true,
+          hasError: false,
+          errorMsg: '',
+        });
+      })
+      .catch((err: Error) => {
+        setinviteTokenForm({
+          submitted: true,
+          submitting: false,
+          success: false,
+          hasError: true,
+          errorMsg: err.message,
+        });
+      });
+  };
+
+  const validateTokenFn = (e: MouseEvent) => {
+    e.preventDefault();
+    checkIsValidToken();
+  };
+
+  const triggerInviteValidation = () => {
+    if (inviteToken?.length && inviteToken?.length === DEFAULT_TOKEN_LENGTH) {
+      checkIsValidToken();
+    }
+  };
+
+  React.useEffect(
+    triggerInviteValidation,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [inviteToken],
+  );
+
+  return (
+    <Box
+      width={_props.isMobile ? '100%' : '38%'}
```

Comment:
Got rid of isMobile, but not sure how it would work with MainAreaCardBox since it doesn't specify any width. Didn't wanna fiddle around with it since it's used in many other places, so just extended/styled a Box.

Reply:
yeah, the required change in the layout-widget is on my branch, I will update it afterwards